### PR TITLE
alexb/refresh data link for signed urls

### PR DESCRIPTION
**Pull request by alex-encord**
A small change: if a data link does not match the pattern for Encord hosting, refresh the data link.
This will ensure that signed URLs are refreshed for private cloud integrations.

### CODE REVIEW

File: src/encord_active/lib/project/project.py [hunk 1]
```diff
@@ -253,13 +256,19 @@ def download_images_from_data_unit(lr: LabelRow, project_file_structure: Project
     lr_structure.images_dir.mkdir(parents=True, exist_ok=True)
     frame_pths: List[Path] = []
     data_units = sorted(lr.data_units.values(), key=lambda du: int(du["data_sequence"]))
+    # If customer uses private cloud integration with signed URLs, refresh data links for label row
+    pat = re.compile(r'^(?!https://storage.googleapis.com/cord-ai-platform.appspot.com/.*)')
+    if any([re.match(pat, du['data_link']) for du in data_units]):
+        lr = encord_project.get_label_row(label_hash)
+        data_units = sorted(lr.data_units.values(), key=lambda du: int(du["data_sequence"]))
```
Comment (frederik-encord) [inline src/encord_active/lib/project/project.py:263]: This seems to defeat the whole purpose of the `download_label_rows`? Now we're essentially downloading every label_row twice?

Comment (frederik-encord) [reply src/encord_active/lib/project/project.py:263]: .. for the cloud integration customers at least.

**Review by frederik-encord — DISMISSED** [review]
I've approved but the re-download seems sub-optimal and we should very soon start refactoring this properly. 

.. Also, what if there are 100 data units? Then we'll only get the first few of them?

File: src/encord_active/lib/project/project.py (hunk 1 above)
Comment (alex-encord) [reply src/encord_active/lib/project/project.py:263]: Yes exactly, this is generally quite a sub optimal fix on top of a sub optimal skeleton. I will make a follow up PR to clean up the entire download flow up as priority was to unblock the customer.

**Review by frederik-encord — APPROVED** [review]

**Pull request merged**